### ISSUE TITLE
ARM Template Failing on SQL Managed Instance - subnet "precondition failed"

### ISSUE THREAD

**Issue by kvietmeier** [microsoft/MCW-Migrating-SQL-databases-to-Azure#14]
Opening new issue because the root cause is different than the other SQLMI issue.

`{
  "code": "DeploymentFailed",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
  "details": [
    {
      "code": "Conflict",
      "message": "{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"**_VnetSubnetConflictWithIntendedPolicy_**\",\r\n        \"message\": \"**_Precondition failed_**.. (https://go.microsoft.com/fwlink/?linkid=871071)\"\r\n      }\r\n    ]\r\n  }\r\n}"
    }
  ]
`}`
`

**Comment by kylebunting** [comment]
@kvietmeier, could you explain the setup process you are using? Are you following the lab guide explicitly and running the ARM template? Or do you have some resources, such as an existing VNet already in place? The subnet conflict error would indicate that the subnet you are trying to use is already in use, so perhaps this could also be an address space conflict? The ARM template uses 10.17.0.0/24 for the subnet prefix.

I just did a test-fix on this lab over the weekend (pull request #13 is pending) and did not run into any issues with the ARM template. I actually ran the ARM template four different times without issue as I was testing. So, I'm assuming there is an environmental issue or conflict happening on your end.

**Comment by kvietmeier** [comment]
Following the Guide explicitly and running the template unaltered.

The only other vnet I have is in another resource group with a 10.10.0.0/16 address space so all of my subnets are 10.10.###.0/24.

I have an FTE Azure account (GBB), so resources shouldn't be an issue.

**Comment by kylebunting** [comment]
Maybe a regional issue? What region are you creating the resources in? Could you try West US 2?

**Comment by kylebunting** [comment]
Is it only the SQL MI deployment that fails with the ARM template? Are the VNet and VNet Gateway deployed correctly? Are the Management and ManagedInstance subnets there?

Could you try to manual deployment steps for SQL MI to see if that works? Trying to determine why the ARM template would fail for you, but not for me. Assuming the other resources are already in place, you should be able to use there during the setup.

Manual steps are documented in the Hands-on lab/Appendix-A.md file. (I've renamed this to Manual-resource-setup.md in the latest pull request, but until that goes in, that's where you'll find the steps).

**Comment by kvietmeier** [comment]
I just kicked of another temples in US East 2 but I can try by hand even if this one goes through.  I'll redo the failed run once this one is done/fails.

**Comment by kvietmeier** [comment]
The Vnet and GW do get created.

**Comment by kylebunting** [comment]
Just to wrap this one up, I'm going to paste in the private email change between @kvietmeier and me.

Karl:
In US East 2 I got much further – but the deployment is stalled. I think might be Region related but not sure why that could be. Tomorrow I will try to walk through it by hand. 

 {
    "error": {
        "code": "GatewayTimeout",
        "message": "The gateway did not receive a response from 'Microsoft.Sql' within the specified time period."
    }
}

![image001](https://user-images.githubusercontent.com/5960229/75555163-b66ca380-5a09-11ea-9676-2cb7666ed107.png)

Kyle:
I’ve run into that before doing SQL MI deployments. Perhaps try deleting your resource group and try again with the ARM template? That has usually worked for me.

From my experience with the product, I feel like there are still some backend deployment and resource issues. Several times, I have had to reach out to support to have them complete a deployment, so I’d say this might be related to the issues there. And I also find that it is difficult to delete a resource group that was the target for SQL MI, even if the deployment fails. It usually takes multiple tries, and several days, to finally delete the resource group when I’m done with the lab.

And please let me know if the manual deployment works.

Karl:
Thanks for responding -
It eventually finished sometime last night. After trying different regions and seeing inconsistent results, I am thinking this isn’t a template issue.
 
I have heard from my colleagues that the SQLmi deployment is inconsistent and can take a very long time. So – likely not a template issue.

Kyle:
Good to hear. I’ll definitely make a note in the workshop to that effect. I already have a note in the “After the hands-on lab” section about issues with deleting the resource group when you are done, but I’ll add another one in the Before the hands-on lab document before the “Deploy to Azure” button for the ARM template.

SQL MI deployment is definitely inconsistent and has typically taken about 4 hours for me when setting up the resources for this lab.

I’ll go ahead and close out the issue in GitHub.
